File: elecciones_app_movil/lib/data/model/candidato/grupo_candidato.dart
import 'package:elecciones_app_movil/data/model/candidato/candidato.dart';
import 'package:elecciones_app_movil/data/model/candidato/movimiento.dart';

class GrupoCandidato {
  GrupoCandidato({
    this.id,
    this.nombre,
    this.movimiento,
    this.fotoArchivo,
    this.fotoNombre,
    this.candidatos,
  });

  int? id;
  String? nombre;
  Movimiento? movimiento;
  dynamic fotoArchivo;
  String? fotoNombre;
  List<Candidato?>? candidatos;

  GrupoCandidato copyWith({
    int? id,
    String? nombre,
    Movimiento? movimiento,
    dynamic fotoArchivo,
    String? fotoNombre,
    List<Candidato?>? candidatos,
  }) =>
      GrupoCandidato(
        id: id ?? this.id,
        nombre: nombre ?? this.nombre,
        movimiento: movimiento ?? this.movimiento,
        fotoArchivo: fotoArchivo ?? this.fotoArchivo,
        fotoNombre: fotoNombre ?? this.fotoNombre,
        candidatos: candidatos ?? this.candidatos,
      );

  factory GrupoCandidato.fromMap(Map<String, dynamic> json) => GrupoCandidato(
        id: json["id"],
        nombre: json["nombre"],
        movimiento: Movimiento?.fromMap(json["movimiento"]),
        fotoArchivo: json["fotoArchivo"],
        fotoNombre: json["fotoNombre"],
        candidatos: json["candidatos"] == null
            ? []
            : json["candidatos"] == null
                ? []
                : List<Candidato?>.from(
                    json["candidatos"]!.map((x) => Candidato.fromMap(x))),
      );

  Map<String, dynamic> toMap() => {
        "id": id,
        "nombre": nombre,
        "movimiento": movimiento,
        "fotoArchivo": fotoArchivo,
        "fotoNombre": fotoNombre,
        "candidatos": candidatos == null
            ? []
            : candidatos == null
                ? []
                : List<dynamic>.from(candidatos!.map((x) => x!.toMap())),
      };
}

File: elecciones-api-cliente/src/main/java/com/eleccciones/cliente/data/entities/candidatos/GrupoCandidato.java
package com.eleccciones.cliente.data.entities.candidatos;

import com.eleccciones.cliente.data.entities.core.impl.AbstractNamed;
import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.Type;

import javax.persistence.*;
import javax.validation.constraints.NotNull;
import java.util.List;

@Entity
@Table(name = "grupo_candidato")
@Getter
@Setter
public class GrupoCandidato extends AbstractNamed {

    @NotNull
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "id_contrato", referencedColumnName = "id")
    @JsonIgnore
    private Contrato contrato;

    @NotNull
    @ManyToOne
    @JoinColumn(name = "id_movimiento", referencedColumnName = "id")
    private Movimiento movimiento;

    @NotNull
    @ManyToOne
    @JoinColumn(name = "id_dignidad_ubicacion", referencedColumnName = "id")
    private DignidadUbicacion dignidadUbicacion;

    @Lob
    @Type(type = "org.hibernate.type.ImageType")
    @Column(name = "foto_archivo")
    private byte[] fotoArchivo;

    @Column(name = "foto_nombre")
    private String fotoNombre;

    @OneToMany(mappedBy = "grupoCandidato")
    private List<Candidato> candidatos;
}
